tests/integration_tests/style/test_rust.py

```diff
@@ -27,34 +30,28 @@ def test_ensure_mod_tests():
 
     @type: style
     """
-    # List all source files containing rust #[test] attribute,
-    # (excluding generated files and integration test directories).
-    # Take the list and check each file contains 'mod tests {', output file
-    # name if it doesn't.
-    cmd = (
-        "/bin/bash "
-        "-c "
-        '"grep '
-        "--files-without-match "
-        "'mod tests {' "
-        "\\$(grep "
-        "--files-with-matches "
-        "--recursive "
-        "--exclude-dir=src/*_gen/* "
-        "'\\#\\[test\\]' ../src/*/src)\" "
-        '| grep -v "../src/io_uring/src/bindings.rs"'
+    excluding = [
+        "_gen/",
+        "/tests/",
+        "/test_utils",
+        "build/",
+        "src/io_uring/src/bindings.rs",
+    ]
+
+    def cmd(string):
+        cmd = shlex.split(string)
+        process = subprocess.run(cmd, shell=True, capture_output=True, check=True)
+        return process.stdout.decode("utf-8").split("\n")
+
+    # Files with `#[test]` without `mod tests`.
+    tests_without_mods = cmd(
+        r'find ../src -type f -name "*.rs" |xargs grep --files-without-match "mod tests {" |xargs grep --files-with-matches "#\[test\]"'
     )
```

Comment:
Don't need the nested function, and I think the `r""` string is messing with shell quoting. Maybe try:

```python
cmd = 'find ../src -type f -name "*.rs" |xargs grep --files-without-match "mod tests {" |xargs grep --files-with-matches "#\[test\]"'
res = subprocess.run(cmd, shell=True, capture_output=True, check=True)
test_without_mods = res.stdout.decode("utf-8").split("\n")
```